### DISCUSSION TITLE
How do I set `make` variables from `nushell`?

### DISCUSSION THREAD

**Discussion by amtoine** [nushell/nushell#6243] [Q&A] [answered]
Hi everyone :wave: :yum: 

## The setup
i've got the following `Makefile`, with a simple `target` which `echo`es the content of the `VAR` variable :ok_hand: 
```rust
$ cat Makefile
VAR=default

target:
    @echo "VAR: ${VAR}"
```

## What happens
all of the following work as expected :ok_hand: 
```rust
$ make target
VAR: default

$ make target VAR=
VAR:

$ make target VAR=foo
VAR: foo
```
i would like to use environment variable inside `VAR`, e.g. the path of the current directory!
but the next commands do not work :thinking: 
```rust
$ make target VAR=$env.PWD
VAR: nv.PWD

$ make target VAR=$"($env.PWD)"
VAR:

$ make target VAR=($env.PWD)
VAR: (nv.PWD)
```
even though `env.PWD` is properly set to the right value, i.e. my `$HOME` here :+1: 
```rust
$ echo $env.PWD
/home/ants
```

## What i wanted to do
I wanted to have a behaviour similar to what can be done in `bash`, i.e.
```bash
$ bash

$ make target VAR=$PWD
VAR: /home/ants
```

## My question
maybe i've missed something in the `string` part of the [`nushell` book](https://www.nushell.sh/book/working_with_strings.html)?
how can i achieve that in `nushell`? :open_mouth: 

anyways, cheers :wink: :tada:

**Comment by kubouch** [comment, marked as answer]
Maybe try make target `$"VAR=($env.PWD)"`. You probably want to send `VAR=/foo/bar/baz` to `make target`, right? From Nushell's point if view it's just a string. It doesn't know that it's supposed to be variable `VAR` with assigned value `/foo/bar/baz`, that's the external command's job to interpret. Our string building is also different from bash etc. so that's why your attempts didn't work.

**Reply by amtoine** [reply]
That's exactly what i was looking for :star_struck: 

thanks a lot :yum:
